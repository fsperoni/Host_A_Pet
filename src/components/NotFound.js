import React from "react";
import oops from '../assets/oops.png'
import '../styles/NotFound.scss';

/**
 * A component to be rendered whenever the URL points to a non-existent route.
 */
const NotFound = () => {
  return (
    <div className="NotFound">
      <h2 className="text-center">Hmmm. I can't seem to find what you want.</h2>
      <img src={oops} alt="Not Found" />
    </div>
  )
};

export default NotFound;