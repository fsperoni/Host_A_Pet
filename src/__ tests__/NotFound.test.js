import { render } from "@testing-library/react";
import NotFound from "../components/NotFound";

// smoke test
it('renders without crashing', function() {
  render(<NotFound />);

});

//snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<NotFound />);
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'Hmmm. I can't seem to find what you want.' in the component.", function() {
  const { getByText } = render(<NotFound />);
  expect(getByText("Hmmm. I can't seem to find what you want.")).toBeInTheDocument();
});