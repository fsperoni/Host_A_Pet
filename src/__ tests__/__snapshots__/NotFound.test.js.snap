// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NotFound"
  >
    <h2
      class="text-center"
    >
      Hmmm. I can't seem to find what you want.
    </h2>
    <img
      alt="Not Found"
      src="oops.png"
    />
  </div>
</DocumentFragment>
`;
